packages/manager/apps/pci-databases-analytics/src/pages/services/create/_components/ApiTerraformDialog.component.tsx

```diff
@@ -0,0 +1,335 @@
+      dialogData.form.handleSubmit(
+        (data) => {
+          const nodesList: TTerraform.cloud.database.service.Node[] = [];
+          // Add nodes to nodesList depending on data.nbNodes
+          for (let i = 0; i < data.nbNodes; i += 1) {
+            nodesList.push({
+              flavor: data.flavor,
+              region: data.region,
+            });
+          }
+
+          const terraformService: TTerraform.cloud.database.ServiceRequest = {
+            body: {
+              description: data.name,
+              nodesList,
+              plan: data.plan,
+              version: `${data.engineWithVersion.version}`,
+              ipRestrictions: data.ipRestrictions,
+            },
+            pathParams: {
+              engine: data.engineWithVersion
+                .engine as TTerraform.cloud.database.EngineEnum,
+              serviceName: projectData.data?.project_id,
+            },
+          };
+          if (dialogData.result.flavor.storage) {
+            terraformService.body.disk = {
+              size:
+                dialogData.result.flavor.storage.minimum.value +
+                data.additionalStorage,
+            } as Disk;
+          }
+          serviceToTerraform(terraformService);
+        },
+        (error) => {
+          console.error(error);
+          toast({
+            title: error,
+            variant: 'destructive',
+          });
+        },
+      )();
+    }
+  };
+
+  useEffect(() => {
+    // Validate form data
+    dialogData.form.handleSubmit(
+      (data) => {
+        // API
+        const apiBody: ServiceCreationWithEngine = {
+          description: data.name,
+          engine: data.engineWithVersion.engine as database.EngineEnum,
+          nodesPattern: {
+            flavor: data.flavor,
+            number: data.nbNodes,
+            region: data.region,
+          },
+          plan: data.plan,
+          version: data.engineWithVersion.version,
+          ipRestrictions: data.ipRestrictions,
+        };
+        if (data.network.type === database.NetworkTypeEnum.private) {
+          const networkOpenstackId = dialogData.result.network.network.regions.find(
+            (r) => r.region.includes(data.region),
+          ).openstackId;
+          apiBody.networkId = networkOpenstackId;
+          apiBody.subnetId = data.network.subnetId;
+        }
+        if (dialogData.result.flavor.storage) {
+          apiBody.disk = {
+            size:
+              dialogData.result.flavor.storage.minimum.value +
+              data.additionalStorage,
+          } as database.service.Disk;
+        }
+        // Destructure body to extract engine and keep the rest
+        const { engine, ...formattedApiBody } = apiBody;
+        setApiEndpoint(
+          `/cloud/project/${projectData.data?.project_id}/database/${engine}`,
+        );
+        setApiBodyContent(formattedApiBody);
+      },
+      (error) => {
+        console.error(error);
+      },
+    )();
+  }, []);
```

Comment:
You don't need useEffect here, as it's a derived state from the props:

```suggestion
   const apiBody: Partial<database.ServiceCreation> = {
    description: dialogData.result.name,
    nodesPattern: {
      flavor: dialogData.result.flavor.name,
      number: dialogData.result.nodes,
      region: dialogData.result.region.name,
    },
    plan: dialogData.result.plan.name,
    version: dialogData.result.version.name,
    ipRestrictions: dialogData.result.ipRestrictions,
  };
  if (dialogData.result.network.type === database.NetworkTypeEnum.private) {
    const networkOpenstackId = dialogData.result.network.network.regions.find(
      (r) => r.region.includes(dialogData.result.region.name),
    ).openstackId;
    apiBody.networkId = networkOpenstackId;
    apiBody.subnetId = dialogData.result.network.subnet.id;
  }
  if (dialogData.result.flavor.storage) {
    apiBody.disk = {
      size:
        dialogData.result.flavor.storage.minimum.value +
        dialogData.result.additionalStorage,
    } as database.service.Disk;
  }
  const apiEndpoint = `/cloud/project/${projectId}/database/${dialogData.result.engine.name}`;
```